cli/command/stack/deploy_composefile.go

```diff
@@ -320,8 +320,20 @@ func deployServices(
 
 			if resolveImage == resolveImageAlways || (resolveImage == resolveImageChanged && image != service.Spec.Labels[convert.LabelImage]) {
 				updateOpts.QueryRegistry = true
+			} else {
```

Comment:
I don't think it should be based on `updateOpts.QueryRegistry`.  `updateOpts.QueryRegistry` is set by comparing the existing state to the desired state of the service.  Setting the Image should be based on that same comparison. `updateOpts.QueryRegistry` is an output, not the source of the state.

>  I'm not entirely convinced using a "switch" is much clearer

I find switch/case reads much better than any if/else, but it's true you could also implement this using if/else if